app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -112,55 +67,43 @@ public void onProviderDisabled(String provider) {
-    private View toolbarView;
     private Button campusButton;
     private Button viewSwitchButton;
     private FloatingActionButton mapCenterButton;
-    //GPS attributes
-    private LocationManager gpsmanager; //LocationManager instance to check gps activity
-    // Search components
-    private SearchView search;
-    private POISearchAdapter poiSearchAdapter;
-    private ExpandableListView searchList;
-    private Dialog searchDialog;
-    private POI location;
-    private POI destination;
-    private SearchState searchState;
+
+    //State
+    private Building lastClickedBuilding;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
 
         parentLayout = (LinearLayoutCompat) inflater.inflate(R.layout.nav_main_fragment, container, false);
-        rootView = parentLayout.findViewById(R.id.navigationMain);
-        toolbarView = parentLayout.findViewById(R.id.nav_toolbar);
 
         //Init Fragments
         transportButtonFragment = (TransportButtonFragment) getChildFragmentManager().findFragmentById(R.id.transportButton);
         indoorMapFragment = (IndoorMapFragment) getChildFragmentManager().findFragmentById(R.id.indoormap);
         directionsFragment = (BottomSheetDirectionsFragment) getChildFragmentManager().findFragmentById(R.id.directionsFragment);
         buildingInfoFragment = (BottomSheetBuildingInfoFragment) getChildFragmentManager().findFragmentById(R.id.buildingInfoFragment);
+
         //Init View Components
         mapCenterButton = (FloatingActionButton) parentLayout.findViewById(R.id.mapCenterButton);
         mapCenterButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (!gpsmanager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
-                    if(alertGPS(mapFragment.getActivity())){
-                        Log.d("Testing AlertGPS Launch", "Finished AlertGPS");
-                    }
-                }
                 if (viewType == ViewType.INDOOR) {
                     viewType = ViewType.OUTDOOR;
                     getChildFragmentManager().beginTransaction().show(mapFragment).hide(indoorMapFragment).commit();
                     getChildFragmentManager().beginTransaction().show(transportButtonFragment).commit(); //To be removed after Outside transportation Google API incorporation
                     campusButton.setVisibility(View.VISIBLE);
                     viewSwitchButton.setText("GO INDOORS");
                 }
-                onClose();
+                ((MainActivity)getActivity()).onClose();
                 //Camera Movement
-                camMove(locateMe(map, mapFragment.getActivity(), gpsmanager, gpsListen));
+                LatLng location = ((MainActivity) getActivity()).getGpsManager().getLocation();
+                if (location != null) {
+                    camMove(location);
```

Comment:
if (location != null){
camMove(location);
map.setMyLocationEnabled(true);
}

Since the method LocateMe was moved out of NavigationFragment, "map.setMyLocationEnabled(true);" must be inserted in this onClickListener here to acquire the "blue dot" from the Google Maps layer.